lib/util/cache/repository/types.ts

```diff
@@ -40,4 +45,9 @@ export interface Cache {
   init?: RepoInitConfig;
   scan?: Record<string, BaseBranchCache>;
   lastPlatformAutomergeFailure?: string;
+  platform?: {
+    github?: {
+      graphqlPageCache?: Record<string, GithubGraphqlPageCache>;
+    };
+  };
```

Comment:
@rarkins So I've altered the cache structure such that it categorizes by platform and functionality, so that we most likely won't have to deal with schema migration and excessive nullability in the leaf interface. However, now it's 3 levels deep.

Reply:
I think it's OK. We won't ever have two platforms at once obviously.. but as you say it can make the typing easier.